__tests__/integration/one-app.spec.js

```diff
@@ -962,6 +955,126 @@ describe('Tests that require Docker setup', () => {
+                    `${appAtTestUrls.cdnUrl}/app/[one-app-version]/service-worker-client.js`,
+                  ]
+                )
+              );
+            });
+
+            test('caches a code-split module chunk and a language pack', async () => {
+              expect.assertions(2);
+
+              await browser.url(`${appAtTestUrls.browserUrl}/demo/franks-burgers`);
+              const openerMessage = await browser.$('#franks-opening-line');
+              await openerMessage.waitForExist();
+              const btn = await browser.$('#order-burger-btn');
+              await btn.click();
+              const franksBurger = await browser.$('#franks-burger');
+              await franksBurger.waitForExist();
+
+              const holocronModuleMap = readModuleMap();
+              const cacheKeys = await browser.executeAsync(getCacheKeys);
+              const cacheMap = new Map(await browser.executeAsync(getCacheEntries, cacheKeys));
+
+              expect(cacheMap.get('__sw/modules')).toHaveLength(4);
+              expect(cacheMap.get('__sw/modules')).toEqual(
+                [
+                  holocronModuleMap.modules['frank-lloyd-root'].browser.url,
+                  holocronModuleMap.modules['preview-frank'].browser.url,
+                  holocronModuleMap.modules['franks-burgers'].browser.url,
+                  // the module chunk
+                  holocronModuleMap.modules['franks-burgers'].browser.url.replace(
+                    'franks-burgers.browser.js',
+                    'Burger.franks-burgers.chunk.browser.js'
+                  ),
+                ]
+              );
+            });
+          });
+
+          describe('invalidation', () => {
+            test('invalidates a language pack if a different locale is loaded', async () => {
+              expect.assertions(2);
+
+              await browser.url(`${appAtTestUrls.browserUrl}/demo/cultured-frankie`);
+
+              const holocronModuleMap = readModuleMap();
+              const localeSelector = await browser.$('#locale-selector');
+              // update the locale
+              await localeSelector.selectByVisibleText('es-MX');
+              await waitFor(500);
+
+              const cacheKeys = await browser.executeAsync(getCacheKeys);
+              const cacheMap = new Map(await browser.executeAsync(getCacheEntries, cacheKeys));
+              // we should not expect an additional lang pack to be added,
+              // rather it replaces the other one used by any given module
+              expect(cacheMap.get('__sw/lang-packs')).toHaveLength(1);
+              expect(cacheMap.get('__sw/lang-packs')).toEqual(
+                [
+                  holocronModuleMap.modules['cultured-frankie'].browser.url.replace(
+                    'cultured-frankie.browser.js',
+                    'es-mx/cultured-frankie.json'
+                  ),
+                ]
+              );
+            });
+          });
+        });
```

Comment:
Can we also test an invalidation cycle for a new version of a Module loaded in addition to language pack?

Reply:
Updated with module version invalidation (05e174b)